store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/utils/TableUtils.java

```diff
@@ -244,13 +245,14 @@ public static Connector getConnectorKerberos(final String instanceName, final St
                 conf.set("hadoop.security.authorization", "true");
                 UserGroupInformation.setConfiguration(conf);
             }
-            // If already logged in with Keytab, then check if ticket needs renewal, else do initial login
-            if (UserGroupInformation.isLoginKeytabBased()) {
+            // If initial login is complete and logged in using Keytab, then check if ticket needs renewal, else do initial login
+            if (initialKerberosLoginComplete && UserGroupInformation.isLoginKeytabBased()) {
```

Comment:
More out of curiosity but would [`isFromKeytab()`](https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/security/UserGroupInformation.html#isFromKeytab--) or [`hasKerberosCredentials()`](https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/security/UserGroupInformation.html#hasKerberosCredentials--) be relevant here/replace this boolean?

Reply:
They wouldn't. It's explained some more in the issue, but briefly the requirement is to ignore any existing login which may have occurred and ensure that a login is done with the credentials supplied in the Gaffer config.

The methods you reference are for instances of UGI, whereas Accumulo uses the static UGI methods. The [`isLoginKeytabBased()`](https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/security/UserGroupInformation.html#isLoginKeytabBased--) is an equivalent.

If a separate keytab login has been done on the system using different credentials prior to starting Gaffer, then the UGI method will return true and login would not happen with the credentials given to Gaffer. As Gaffer is given creds they should always be used for establishing the connection, that's what this new boolean ensures.

Reply:
Nice that makes sense to ignore any existing login